src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -1498,11 +1586,42 @@ public void Dispose()
             // Do this here because otherwise if a purge job runs there will
             // be a deadlock while the purge job waits for the lock that is held
             // by the disposal which is waiting for timer callbacks to finish
-            StopExpirePurgeTimer(); 
-            ThreadSafety.DoLocked(() =>
-            {
-                ThrowIfActiveItems();
-                Dispose(true);
+            StopExpirePurgeTimer();
+            ThreadSafety.DoLocked(() => {
+                if (IsClosed) {
+                    return;
+                }
+
+                if (!_isClosing) {
+                    _isClosing = true;
+                }
+            });
+
+            foreach (var q in ActiveLiveQueries) {
+                q.Key.Stop();
+            }
+
+            while (!IsReadyToClose) {
+                _closeCondition.WaitOne();
+                _closeCondition.Reset();
+            }
+
+            ThreadSafety.DoLocked(() => {
+                if (IsClosed) {
+                    return;
+                }
+
+                ClearUnsavedDocsAndFreeDocObservers();
+                FreeC4DbObserver();
```

Comment:
What is the difference between here and `Dispose(bool)` which appears to largely do the same stuff?  This will cause confusion and problems later if you have to maintain two places that are doing the same things.